enarx-copy-labels-linked

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+from github import Github
+import itertools
+import os
+import re
+
+# We don't want to copy all labels on linked issues; only those in this subset.
+BLOCKED_LABELS = {
+    "meta",
+    "Polls",
+    "question",
+    "wontfix",
+    "help wanted",
+    "good first issue",
+    "duplicate",
+}
+
+# Get all issue numbers related to a PR.
+def get_related_issues(pr):
+    # Regex to pick out closing keywords.
+    regex = re.compile("(close[sd]?|fix|fixe[sd]?|resolve[sd]?)\s*:?\s+#(\d+)", re.I)
+
+    # Extract all associated issues from closing keyword in PR
+    for verb, num in regex.findall(pr.body):
+        yield int(num)
+
+    # Extract all associated issues from PR commit messages
+    for c in pr.get_commits():
+        for verb, num in regex.findall(c.commit.message):
+            yield int(num)
+
+# Cache all project cards in a dictionary for easy lookup (and to avoid
+# redundant network requests).
+def cache_board(project):
+    card_cache = {}
+    for column in project.get_columns():
+            for card in column.get_cards():
+                content = card.get_content()
+                if content:
+                    card_cache[content.id] = (card, column)
+    return card_cache
+
+github = Github(os.environ['GITHUB_TOKEN'])
+
+# Get the Enarx org.
+org = github.get_organization('enarx')
+
+# Get the Planning board.
+planning = {p.name: p for p in org.get_projects(state='open')}['Planning']
+
+# Cache the Planning board to avoid redundant network requests.
+planning_cached = cache_board(planning)
+
+for issue in github.search_issues(f"org:enarx is:pr is:public is:open linked:issue -is:draft project:enarx/{planning.number}"):
```

Comment:
If a PR wasn't on the sprint board, it would cause problems with the other action (from https://github.com/enarx/bot/pull/14) attempting to move a card to a column not in the same project. However, with them separated, it's probably not an issue to update the query to your proposal.